futures-util/src/async_await/random.rs

```diff
@@ -0,0 +1,51 @@
+use std::{
+    cell::Cell,
+    collections::hash_map::DefaultHasher,
+    hash::Hasher,
+    num::Wrapping,
+    sync::atomic::{AtomicUsize, Ordering},
+};
+
+#[doc(hidden)]
+pub fn shuffle<T>(slice: &mut [T]) {
```

Comment:
Maybe it would be worth linking this to https://en.m.wikipedia.org/wiki/Fisher–Yates_shuffle